### ISSUE TITLE
Order of Shape's color cycle in examples is confusing

### ISSUE THREAD

**Issue by andy-sweet** [napari/napari#3013]
## 🐛 Bug

The order of the colors generated from a color cycle in a couple of shape's examples is confusing.
- [`add_shapes_with_properties.py`](https://github.com/napari/napari/blob/master/examples/add_shapes_with_properties.py)
- [`add_shapes_with_text.py`](https://github.com/napari/napari/blob/master/examples/add_shapes_with_text.py).

In both examples, a color is defined via a property name (`'class'`) and the associated color cycle is specified as a list of named colors (`['blue', 'magenta', 'green']`). But the property actually stores class names (e.g. `['sky', 'person', 'building']`), so it's unclear how that helps define a color. I assume that the order of shapes and the order of the color cycle would correspond, but that doesn't happen.

## To Reproduce

See the examples. But also, here's a minimal example that demonstrates my confusion.

```python
import napari
import numpy as np

viewer = napari.Viewer()
data = np.random.rand(3, 4, 2)
properties = {'class': ['hand', 'face', 'camera']}
class_colors = ['blue', 'magenta', 'green']
shapes = viewer.add_shapes(data, shape_type='polygon', properties=properties, edge_color='class', edge_color_cycle=class_colors)
print(shapes.edge_color)
```

Note that `shapes.edge_color` is effectively `['green', 'magenta', 'blue']`. I think that's because the edge_colors end up being generated according to the sorted order of the `'class'` property's values (see [`Shapes._map_color`](https://github.com/napari/napari/blob/0f1b6809cf583cb7e049ddcb32fe252afb3747a6/napari/layers/shapes/shapes.py#L1336).

## Expected behavior

I'm not sure because it's strange to specify the color via a property that doesn't store color values. But my guess is that the order of the color cycle should be maintained and the non-color property should effectively be ignored. Otherwise, we should change the examples to directly specify the colors or use a property that specifies colors.

## Environment

napari: 0.4.9rc1.dev70+g5fe37b47.d20210709
Platform: macOS-11.3.1-x86_64-i386-64bit
System: MacOS 11.3.1
Python: 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:13:15) [Clang 11.1.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.19.5
SciPy: 1.6.3
Dask: 2021.04.1
VisPy: 0.6.6

OpenGL:
- GL version: 2.1 INTEL-16.2.16
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- brainglobe-io: 0.0.2
- brainreg-standard: 0.0.2
- cellfinder: 0.0.10
- console: 0.0.3
- napari-demo: 0.0.2
- napari-properties-plotter: 0.1.0
- napari-properties-viewer: 0.0.2
- napari-skimage-regionprops: 0.1.0
- ndtiffs: 0.1.1
- scikit-image: 0.4.9rc1.dev70+g5fe37b47.d20210709
- svg: 0.1.5

## Additional context

I noticed this when adding an example in [PR #2969](https://github.com/napari/napari/pull/2969).

@kevinyamauchi : let me know if I misunderstood the usage in the examples. If I didn't and this needs to be fixed, then I think one way to do it may be to use `ColorManager` in `Shapes` because a similar example with `Points` does not have the same issue. That's probably something we want to do anyway too.

**Comment by kevinyamauchi** [comment]
@andy-sweet , thanks for flagging this! You are correct. The intended behavior should be for the cycles to be cycled in the order of the data (i.e., hand should be blue, face should be magenta, and camera should be green). I agree that it makes sense to fix this by integrating`ColorManager` into `Shapes`.

**Comment by andy-sweet** [comment]
Thanks for clarifying @kevinyamauchi!. I'll create a separate issue for integrating `ColorManager` into `Shapes` to make sure that work isn't repeated and will link it to this issue.

**Comment by willingc** [comment]
Closing this issue as addressed by opening #3026
